client/src/components/RelatedItems/RelatedItems.jsx

```diff
@@ -46,27 +44,53 @@ function RelatedItems() {
           .catch((err) => console.error(err))
       )
     ).then((res) => setRelatedItemsDetails(res));
+
+    Promise.all(
+      relatedItems.map((product) =>
+        axios
+          .get(`/reviews/?product_id=${product}`)
+          .then((response) => response.data)
+          .catch((err) => console.error(err))
+      )
+    ).then((res) => setRelatedReviews(res));
   }, [relatedItems]);
 
   useEffect(() => {
+    const aveRating = _.map(relatedReviews, (product) => ({
+      aveRating:
+        _.reduce(product.results, (sum, num) => (sum + num.rating), 0) /
+        product.results.length,
+    }));
+
     const accArr = _.map(relatedItems, (ea, index) =>
-      _.extend(relatedItemsImg[index], relatedItemsDetails[index])
+      _.extend(
+        {},
+        relatedItemsImg[index],
+        relatedItemsDetails[index],
+        aveRating[index],
+      )
     );
     setJoinedAPIDetails(accArr);
-  }, [relatedItemsDetails]);
+  }, [relatedReviews]);
 
   useEffect(() => {
     if (yourOutfitId) {
       const getYOImages = axios.get(`/products/${yourOutfitId}/styles`);
       const getYODetails = axios.get(`/products/${yourOutfitId}`);
+      const getYOReviews = axios.get(`/reviews/?product_id=${yourOutfitId}`)
       axios
-        .all([getYOImages, getYODetails])
-        .then((response) =>
+        .all([getYOImages, getYODetails, getYOReviews])
+        .then((response) => {
+          const aveRating = {
+            aveRating: _.reduce(response[2].data.results, (sum, num) => (
```

Comment:
I'm leaving a note here so I don't forget:
Can you explain your thought process on this section of code?